### PR TITLE
Safe directory resolution

### PR DESCRIPTION
**Pull request by cjllanwarne**

### CODE REVIEW

File: src/main/scala/wdl4s/WdlNamespace.scala [hunk 1]
```diff
@@ -525,19 +524,34 @@ object WdlNamespace {
 
   def fileResolver(str: String): WdlSource = readFile(Paths.get(str))
 
-  def directoryResolver(directory: File)(str: String): WdlSource = fileResolver(directory.path.resolve(str).toString)
+  def directoryResolver(directory: File)(str: String): WdlSource = {
+    val absolutePath = directory.path.resolve(str).toAbsolutePath
+    if (absolutePath.startsWith(directory.path.toAbsolutePath)) {
```
Comment (Horneth) [inline src/main/scala/wdl4s/WdlNamespace.scala]: Would a `str` like "../../../../etc/password" be handle properly ?

File: src/main/scala/wdl4s/WdlNamespace.scala [hunk 2]
```diff
@@ -525,19 +524,34 @@ object WdlNamespace {
 
   def fileResolver(str: String): WdlSource = readFile(Paths.get(str))
 
-  def directoryResolver(directory: File)(str: String): WdlSource = fileResolver(directory.path.resolve(str).toString)
+  def directoryResolver(directory: File)(str: String): WdlSource = {
```
Comment (cjllanwarne) [inline src/main/scala/wdl4s/WdlNamespace.scala]: In my imports branch, Cromwell calls into this one directly to get the resolver

**Comment by ruchim** [comment]
👍  safety first

**Comment by Horneth** [comment]
👍

**Pull request merged**